### ISSUE TITLE
Can't return value from closure in `Client.time`

### ISSUE THREAD

**Issue by lilydjwg** [markstory/rust-statsd#7]
Client.time is declared as

```
fn time<F>(&mut self, metric: &str, callable: F) where F: Fn()
```

So it can't be used to run a function that returns a value. It'd be better to be:

```
fn time<F, R>(&mut self, metric: &str, callable: F) -> R where F: Fn() -> R
```

**Comment by markstory** [comment]
Thank! I'll get that fixed up.

**Comment by markstory** [comment]
Fixed in 0.7.0
